agent/SlapdConfigAgent.cc

```diff
@@ -422,11 +434,38 @@ YCPValue SlapdConfigAgent::Execute( const YCPPath &path,
             attrs.add("contextCSN");
             LDAPSearchResults *sr = m_lc->search( "cn=config", LDAPConnection::SEARCH_SUB,
                                                   "objectclass=*", attrs );
+            std::vector<LDAPEntry> searchResult;
+            while (LDAPEntry *e = sr->getNext())
+            {
+                searchResult.push_back(LDAPEntry(*e));
+            }
+            // Explicitly load syncprov.so and DB drivers if modules should be loaded explicitly
+            OlcModuleListEntry moduleListEntry;
+            if (moduleLoadPath != "")
+            {
+                for (std::vector<LDAPEntry>::iterator it = searchResult.begin(); it < searchResult.end(); it++)
+                {
+                    // Look for OlcModuleList and remove its LDAPEntry from result
```

Comment:
Huh? I don't understand what is going on, but this is suspicious.
The comment explains *what*  is being done. But anyone can understand that by looking at the next 4 lines. Instead, explain *why* it needs to be done.